src/Compilers/Core/Portable/Syntax/GreenNode.NodeFlagsAndSlotCount.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal abstract partial class GreenNode
+    {
+        /// <summary>
+        /// Combination of <see cref="NodeFlags"/> and <see cref="SlotCount"/> stored in a single 16bit value.
+        /// </summary>
+        private struct NodeFlagsAndSlotCount
+        {
+            /// <summary>
+            /// 4 bits for the SlotCount.  This allows slot counts of 0-14 to be stored as a direct byte.  All 1s
+            /// indicates that the slot count must be computed.
+            /// </summary>
+            private const ushort SlotCountMask = 0b1111000000000000;
+            private const ushort NodeFlagsMask = 0b0000111111111111;
+
+            private const int SlotCountShift = 12;
+            private const int MaxSlotCount = 15;
```

Comment:
`Max` here implies that this is actually the maximum, but 15 is not actually the max count. The max number we can actually represent is actually 14. I would actually call this `ShiftedSlotMask` and assign it `0b0000000000001111` as a value, to make it clear what you're actually trying to do.

Reply:
Talked offline as well: I'd like to use this constant directly in `GreenNode`, rather than adjusting the return to `byte.MaxValue`, to make the logic clearer.

Reply:
Done.